tfx_addons/sampler/executor_test.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executor test for the sampling component's executor."""
+
+import filecmp
+import os
+import random
+import tempfile
+
+import apache_beam as beam
+import tensorflow as tf
+from absl.testing import absltest
+from apache_beam.testing.util import assert_that, equal_to
+from tfx.components.util import tfxio_utils
+from tfx.dsl.io import fileio
+from tfx.types import artifact_utils, standard_artifacts
+from tfx.utils import io_utils, json_utils
+
+from tfx_addons.sampler import executor, spec
+
+
+class ExecutorTest(absltest.TestCase):
+  def _validate_output(self, output, splits, num=1):
+    def generate_elements(data):
+      for i in range(len(data[list(data.keys())[0]])):
+        yield {
+            key:
+            data[key][i][0] if data[key][i] and len(data[key][i]) > 0 else ""
```

Comment:
Consider breaking this up and naming parts of it for the reader, fairly complex data generation here